lib/editor/components/ExceptionDateRange.js

```diff
@@ -0,0 +1,191 @@
+  }
+
+  _onSelect = (key: string) => {
+    const {activeComponent, activeEntity, range, updateActiveGtfsEntity} = this.props
+    let dates = [...activeEntity.dates]
+
+    const newDates = activeEntity.dates.filter(date => typeof date === 'number')
+    // $FlowFixMe Flow does not recognize filtering on typeof
+    dates = (dates.filter(date => typeof date === 'string'): Array<string>)
+
+    const startMoment = moment(range.startDate)
+    const endMoment = moment(range.endDate)
+    const deleteStart = key === 'range' ? startMoment : startMoment.add(1, 'days') // Preserve the first day if deleting only the end date
+
+    dates = modifyRangeOfDates(dates, endMoment, deleteStart, true)
+
+    // We need to sort the dates before update so that indices are maintained for valdiation.
+    sortDates(dates)
+
+    updateActiveGtfsEntity({
+      component: activeComponent,
+      entity: activeEntity,
+      props: {dates: [...dates, ...newDates]}
+    })
+  }
+
+  render () {
+    const {index, range, validationErrors, validationState} = this.props
+    const endMoment = range.endDate ? moment(range.endDate) : moment(range.startDate).add(1, 'days')
+    const startMoment = range.startDate ? moment(range.startDate) : undefined
+
+    const dateTimeProps = {
+      defaultText: undefined,
+      mode: 'date',
+      size: 'sm',
+      style: {gridColumn: 'span 5'}
+    }
+
+    return (
+      // Validation state override allows for control of input before date
+      <FormGroup
+        key={index}
+        style={{
+          display: 'grid',
+          gridGap: '5px',
+          gridTemplateColumns: 'repeat(12, 1fr)',
+          marginBottom: '5px',
+          textAlign: 'center'
+        }}
+        validationState={validationState === 'error' || this.state.forceErrorState ? 'error' : 'success'}
+      >
+        <div
+          className='exception-date-picker'
+          data-test-id={`exception-date-range-${index}-1`} // First range input
+          style={{gridColumn: 'span 5'}}
+        >
+          <DateTimeField
+            className='date-range'
+            dateTime={startMoment}
+            inputProps={{style: inputStyleProps}}
+            // The maxDate API subtracts a day from the maxDate, see:
+            // https://github.com/quri/react-bootstrap-datetimepicker/commit/eb5f700267d0cf945abfdac3a4f7ca879b314e70
+            maxDate={endMoment.clone().subtract(1, 'day')}
+            onChange={(millis) => this._onRangeUpdate(millis, true)}
+            {...dateTimeProps}
+          />
+        </div>
+        <i className='fa fa-minus' style={{gridColumn: 'span 1', placeSelf: 'center '}} />
+        <div
+          className='exception-date-picker'
+          data-test-id={`exception-date-range-${index}-2`} // Second range input
+          style={{gridColumn: 'span 5'}}
+        >
+          <DateTimeField
+            className='date-range'
+            dateTime={endMoment}
+            inputProps={{style: dateTimeProps}}
```

Comment:
Is this the right prop?

Reply:
Good catch! [https://github.com/ibi-group/datatools-ui/pull/1032/commits/b3e3359168c9e32406bca20cca06b8d8a242c68f]